src/lib/parser.c

```diff
@@ -372,3 +363,62 @@ parser_error_t parser_getItem(parser_context_t *ctx,
     const int8_t displayDynIdx = displayIdx - numberFixedItems;
     return parser_getDynamicItem(ctx, displayDynIdx, outKey, outKeyLen, outVal, outValLen, pageIdx, pageCount);
 }
+
+__Z_INLINE parser_error_t parser_getItemEntity(parser_context_t *ctx,
+                                           int8_t displayIdx,
+                                           char *outKey, uint16_t outKeyLen,
+                                           char *outVal, uint16_t outValLen,
+                                           uint8_t pageIdx, uint8_t *pageCount) {
+
+    if (displayIdx == 0) {
+        snprintf(outKey, outKeyLen, "ID");
+        return parser_printPublicKey(&parser_tx_obj.oasis_entity.id,
+                                     outVal, outValLen, pageIdx, pageCount);
+    }
+
+    if (displayIdx <= (int) parser_tx_obj.oasis_entity.nodes_length) {
+        const int8_t index = displayIdx -1;
+
+        snprintf(outKey, outKeyLen, "Node [%i]", index);
+
+        parser_error_t err = _getNodesIdAtIndex(ctx, &parser_tx_obj, index);
+        if (err != parser_ok)
+            return err;
+
+        return parser_printPublicKey(&parser_tx_obj.oasis_entity.node,
+                                     outVal, outValLen, pageIdx, pageCount);
+    }
+
+    if (displayIdx - parser_tx_obj.oasis_entity.nodes_length == 1) {
+        if (parser_tx_obj.oasis_entity.allow_entity_signed_nodes) {
+            snprintf(outKey, outKeyLen, "Allowed");
+        } else {
+            snprintf(outKey, outKeyLen, "Not Allowed");
+        }
+        return parser_ok;
+    }
+
+}
+
+parser_error_t parser_getItem(parser_context_t *ctx,
+                              int8_t displayIdx,
+                              char *outKey, uint16_t outKeyLen,
+                              char *outVal, uint16_t outValLen,
+                              uint8_t pageIdx, uint8_t *pageCount) {
+    MEMZERO(outKey, outKeyLen);
+    MEMZERO(outVal, outValLen);
+    snprintf(outKey, outKeyLen, "?");
+    snprintf(outVal, outValLen, " ");
+
+    if (displayIdx < 0 || displayIdx >= parser_getNumItems(ctx)) {
+        return parser_no_data;
+    }
+
+    // REVIEW: can't verify type in C
```

Comment:
not a reliable one. I think the suggestion of using an enum to specify the struct embeded in the union would be an acceptable solution